src/Microsoft.AspNet.Mvc.Razor/RazorViewEngine.cs

```diff
@@ -228,6 +228,29 @@ public virtual IEnumerable<string> AreaViewLocationFormats
             return new RazorPageResult(pageName, searchedLocations);
         }
 
+        private string GetNormalizedRouteValue(string key, ActionContext context)
+        {
+            if (context.ActionDescriptor.AttributeRouteInfo == null)
+            {
+                var match = context.ActionDescriptor.RouteConstraints.SingleOrDefault(r => r.RouteKey == key);
+                if (match != null)
+                {
+                    return match.RouteValue;
+                }
+            }
+            else
+            {
+                object match;
+                context.ActionDescriptor.RouteValueDefaults.TryGetValue(key, out match);
+                if (match != null)
+                {
+                    return match.ToString();
+                }
+            }
+
+            return context.RouteData.Values.GetValueOrDefault<string>(key);
```

Comment:
should we compare the route data **value** against the AD value case-insensitively as a check? I can't think of any super-valid case where this matters, but it will make sure we're not doing more than we intended to.